homeassistant/components/legrand_rflc/light.py

```diff
@@ -0,0 +1,180 @@
+"""The Legrand RFLC integration light platform."""
+
+from collections.abc import Mapping
+from typing import Final
+
+from homeassistant import config_entries
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_TRANSITION,
+    COLOR_MODE_BRIGHTNESS,
+    SUPPORT_TRANSITION,
+    LightEntity,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+
+class _LegrandRflcSwitch(LightEntity):
+    _attr_should_poll = False
+
+    def __init__(self, hub, zid: int, properties: Mapping):
+        self._hub = hub
+        self._attr_name = properties[hub.NAME]
+        self._attr_is_on = properties[hub.POWER]
+        self._zid = zid
+        hub.on(
+            f"{hub.EVENT_ZONE_PROPERTIES_CHANGED}:{zid}",
+            self._zone_properties_changed,
+        )
+        hub.on(hub.EVENT_CONNECTED, self._available)
+        hub.on(hub.EVENT_DISCONNECTED, self._available)
+        hub.on(hub.EVENT_AUTHENTICATED, self._available)
+        hub.on(hub.EVENT_UNAUTHENTICATED, self._available)
+
+    @property
+    def unique_id(self) -> str:
+        return f"{self._hub.host()}:{self._zid}"
+
+    @property
+    def available(self) -> bool:
+        return self._hub.connected and self._hub.authenticated
+
+    async def _available(self) -> None:
+        self.async_write_ha_state()
+
+    def _zone_properties_changed_switch(self, message: Mapping) -> None:
+        hub = self._hub
+        if hub.PROPERTY_LIST in message:
+            properties = message[hub.PROPERTY_LIST]
+            if hub.POWER in properties:
+                self._attr_is_on = properties[hub.POWER]
+
+    async def _zone_properties_changed(self, message: Mapping) -> None:
+        self._zone_properties_changed_switch(message)
+        self.async_write_ha_state()
+
+    async def _async_switch(self, power: bool) -> None:
+        hub = self._hub
+
+        async def handle(message: Mapping):
+            hub.StatusError(message).raise_if()
+
+        await hub.handle_send(
+            handle, hub.compose_set_zone_properties(self._zid, power=power)
+        )
+
+    async def async_turn_on(self, **kwargs) -> None:
+        await self._async_switch(True)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        await self._async_switch(False)
+
+
+class _LegrandRflcDimmer(_LegrandRflcSwitch):
+    _attr_color_mode = COLOR_MODE_BRIGHTNESS
+    _attr_supported_color_modes = {COLOR_MODE_BRIGHTNESS}
+    _attr_supported_features = SUPPORT_TRANSITION
+
+    @staticmethod
+    def _normalize(value: int, ceiling: int) -> float:
+        """Normalize [0, ceiling] to [0.0, 1.0]."""
+        return value / ceiling
+
+    @staticmethod
+    def _quantize(value: float, ceiling: int) -> int:
+        """Quantize [0.0, 1.0] to [0, ceiling]."""
+        return int(value * ceiling)
```

Comment:
> We should round here instead of truncate. Also, note that Home Assistant brightness 0 means the light is off, and Home Assistant brightness 1 means the light is at it's minimum brightness. Is that compatible with the legrand API?

Yes, Legrand RFLC considers a brightness of 0 as off.
https://github.com/home-assistant/core/pull/53430/commits/6e0d55720c8b21686da5f58fc5f6e90a5d2d4249